src/store/index.js

```diff
@@ -0,0 +1,29 @@
+import Vue from "vue";
+import Vuex from "vuex";
+import Http from "../api/apiInterface.js";
+
+Vue.use(Vuex);
+
+const store = new Vuex.Store({
+  state: {
+    productList: [],
+  },
+  mutations: {
+    setProductList(state, payload) {
+      state.productList = payload;
+    },
+  },
+  actions: {
+    async fetchProductList(context) {
+      const data = await Http.get("/products");
```

Comment:
Consider setting the data list to empty before fetching to avoid showing stale/incorrect data on endpoint error.
Also, a nitpick consider handling error. 😉